internal/controller/controllers/preprovisioningimage_controller.go

```diff
@@ -181,28 +181,44 @@ func (r *PreprovisioningImageReconciler) Reconcile(origCtx context.Context, req
 		return ctrl.Result{Requeue: true, RequeueAfter: time.Until(imageTimePlusCooldown)}, nil
 	}
 
-	patch := client.MergeFrom(image.DeepCopy())
-	imageUpdated := image.Status.ImageUrl != "" && image.Status.ImageUrl != infraEnv.Status.ISODownloadURL
-	err = r.setImage(image, *infraEnv)
+	return ctrl.Result{}, r.handleImageUpdate(ctx, log, image, infraEnv)
+}
+
+func (r *PreprovisioningImageReconciler) handleImageUpdate(ctx context.Context, log logrus.FieldLogger, image *metal3_v1alpha1.PreprovisioningImage, infraEnv *aiv1beta1.InfraEnv) error {
+	bmh, err := r.getBMH(ctx, image)
```

Comment:
Do you think there would be an issue if the BMH no longer exists? Like would it continuously reconcile?

Reply:
Gotcha! Just wanted to make sure we didn't end up in an endless reconcile here